### PR TITLE
Ports: Add open-adventure

### PR DESCRIPTION
**Pull request by ninadsachania**

### CODE REVIEW

File: Ports/open-adventure/patches/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+Use [`editline`](../../editline/) instead of `libedit`.
+
+## `0002-Change-headers-for-libeditline.patch`
+
+Change headers because we are now using [`editline`](../../editline/).
+
+## `0003-Rearrange-headers-because-editline.h-uses-FILE.patch`
+
+Move `stdio.h` before `editline.h` because it uses `FILE`.
```
Comment (timschumi) [inline Ports/open-adventure/patches/ReadMe.md:13]: Please merge those three patches, they all are related to using libeditline instead of libedit.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/open-adventure/patches/ReadMe.md (hunk 1 above)
Comment (ninadsachania) [reply Ports/open-adventure/patches/ReadMe.md]: Done.

File: Ports/open-adventure/package.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='open-adventure'
+version='1.20'
+files=(
+    "https://gitlab.com/esr/open-adventure/-/archive/1.20/open-adventure-${version}.zip#441637afaf4f7db0e96b3d27ab8fb17ca68e0df343c72a9e4416569de1b4ea50"
+)
+
+depends=(
+    "editline"
```
Comment (timschumi) [inline Ports/open-adventure/package.sh:9]: Not an interpolated string, so single quotes.

File: Ports/open-adventure/package.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+bin_path="/usr/local/bin"
```
Comment (timschumi) [inline Ports/open-adventure/package.sh:12]: Same here.

File: Ports/open-adventure/patches/ReadMe.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+## `0001-Use-libeditline-instead-of-libedit.patch`
+
+Three changes:
+
+1. Change `libedit` to [`editline`](../../editline/) in the `Makefile`.
+
+2. Change `#include <editline/readline.h>` to `#include <editline.h>`.
+
+3. Move `stdio.h` before `editline.h` because it uses `FILE`.
```
Comment (timschumi) [inline Ports/open-adventure/patches/ReadMe.md:11]: This does not match the commit message in the actual patch. Is this file properly autogenerated?

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/open-adventure/patches/ReadMe.md (hunk 4 above)
Comment (ninadsachania) [reply Ports/open-adventure/patches/ReadMe.md:11]: No. I wrote that. I didn't know it was supposed to be autogenerated. I don't see this convention being mentioned anywhere in the docs. I'll change it.

File: Ports/open-adventure/package.sh (hunk 3 above)
Comment (ninadsachania) [reply Ports/open-adventure/package.sh]: Done.

File: Ports/open-adventure/package.sh (hunk 2 above)
Comment (ninadsachania) [reply Ports/open-adventure/package.sh]: Done.

File: Ports/open-adventure/patches/ReadMe.md (hunk 4 above)
Comment (ninadsachania) [reply Ports/open-adventure/patches/ReadMe.md]: Done.

**Comment by ninadsachania** [comment]
Thanks for reviewing! :^)

Question: What's the point of the `ReadMe.md` file if it repeats the information that is already present in the `.patch` files. Just curious.

**Comment by timschumi** [comment]
> Question: What's the point of the `ReadMe.md` file if it repeats information that is already present in the `.patch` files. Just curious.

I'll admit that a dedicated README was more useful back when the patch format wasn't standardized yet, but it still is a good way of getting an overview over what has been changed (and ocasionally also _why_).

**Review by timschumi — APPROVED** [review]
Took care of some remaining formatting issues, otherwise good as-is.

**Pull request merged**